### PR TITLE
Update 2021.8.0

### PR DESCRIPTION
**Pull request by danpetry**
# Links

[Upstream](https://github.com/oneapi-src/oneTBB)
[Upstream update diff](https://github.com/oneapi-src/oneTBB/compare/v2021.7.0...v2021.8.0)
[Conda-forge update diff](https://github.com/conda-forge/tbb-feedstock/compare/e077d70432771162228359c4c9fc81fb987c081b..435de0a0ea2f111ee49d7ba6bd308afd926eb13a)
[Changelog](https://github.com/oneapi-src/oneTBB/blob/master/RELEASE_NOTES.md)
[Jira ticket](https://anaconda.atlassian.net/browse/PKG-1344)
[License](https://github.com/oneapi-src/oneTBB/blob/master/LICENSE.txt)

# Changes
- Bump version, update checksum
- Change pin_subpackage to an explicit pinning to solve unresolvability error during tbb4py testing. `Pin_subpackage(exact=True)` uses the package hash, which will never be resolveable with a
pinning specified using an equality.
- lint addressing:
  - Rename top-level package so it's different from the main package name
  - Add requirements to main output, copied from top-level package
  - Move main output test to main output section, not top-level section
  - Add description
  - Update doc_url, previous was a redirect to docs for an older version
  - Remaining lints are erroneous and have been reported as issues
- other cleanup:
  - Remove unneeded git and python dependencies

# Further notes
- No changes to dependencies or build system interface upstream. No relevant changes in conda-forge either.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -32,34 +32,24 @@
 {% set tbb4py_build_dir = 'build_py%PY_VER%h%PKG_HASH%' %}  # [win]
 
 package:
-  name: tbb
+  name: tbb-split
   version: {{ version }}
 
 source:
   fn: {{ vtag }}.tar.gz
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:39]: `  fn: {{ vtag }}.tar.gz` can be removed, it's obsolete

**Review by skupr-anaconda — APPROVED** [review]
LGTM

File: recipe/meta.yaml [hunk 2]
```diff
@@ -160,8 +164,11 @@ about:
     - LICENSE.txt
     - third-party-programs.txt
   summary: High level abstract threading library
+  description: |
+    oneTBB is a flexible C++ library that simplifies the work of adding parallelism to complex applications, even if you
+    are not a threading expert.
   dev_url: https://github.com/oneapi-src/oneTBB
-  doc_url: https://software.intel.com/en-us/oneapi-tbb-documentation
+  doc_url: https://www.intel.com/content/www/us/en/developer/tools/oneapi/onetbb.html
```
Comment (cbouss) [inline recipe/meta.yaml:171]:
```suggestion
  doc_url: https://oneapi-src.github.io/oneTBB/
```

The github project points to this link.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -83,7 +86,6 @@ outputs:
         - tbb >={{ version }}
     requirements:
       build:
-        - python
         - {{ compiler('cxx') }}
         - ninja  # [win]
         - make   # [not win]
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml:91]: just a question of curiosity can we use ninja for all platforms ?

Comment (cbouss) [reply recipe/meta.yaml:91]: We could, yes.

File: recipe/meta.yaml (hunk 1 above)
Comment (danpetry) [reply recipe/meta.yaml:39]: done

File: recipe/meta.yaml [hunk 4]
```diff
@@ -139,7 +141,9 @@ outputs:
     test:
       requires:
         - python
-        - {{ pin_subpackage('tbb', exact=True) }}        # we want to test with this specific tbb package
+        - tbb {{ version }}                                # we want to test with this specific tbb package.
```
[the remark this answers is not in the document]
Comment (danpetry) [reply recipe/meta.yaml:144]: yeah true. ok, sounds fine

File: recipe/meta.yaml (hunk 2 above)
Comment (danpetry) [reply recipe/meta.yaml:171]: great, thanks

File: recipe/meta.yaml (hunk 3 above)
Comment (danpetry) [reply recipe/meta.yaml]: thanks, nice one

**Comment by skupr-anaconda** [comment]
Looks like there are missing cmake flags for linux platforms https://github.com/AnacondaRecipes/tbb-feedstock/pull/9/files#diff-f3725a55bf339595bf865fec73bda8ac99f283b0810c205442021f29c06eea9aR18-R24

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

**Pull request merged**